CLI/controllers/interact_test.go

```diff
@@ -0,0 +1,195 @@
+package controllers_test
+
+import (
+	cmd "cli/controllers"
+	mocks "cli/mocks/controllers"
+	"cli/models"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func interactTestSetup(t *testing.T) (cmd.Controller, *mocks.APIPort, *mocks.Ogree3DPort) {
+	controller, mockAPI, mockOgree3D := layersSetup(t)
+
+	mockGetObject(mockAPI, createRoom)
+
+	mockCreateObject(mockAPI, "generic", map[string]any{
+		"name":        "A01",
+		"category":    "generic",
+		"description": []any{},
+		"domain":      createRoom["domain"],
+		"parentId":    createRoom["id"],
+		"attributes": map[string]any{
+			"height":       "1",
+			"heightUnit":   "cm",
+			"rotation":     `{"x":0, "y":0, "z":0}`,
+			"posXYZ":       `{"x":1 ,"y":1 ,"z":1 }`,
+			"posXYUnit":    "m",
+			"size":         `{"x":1 ,"y":1 }`,
+			"sizeUnit":     "cm",
+			"shape":        "cube",
+			"type":         "box",
+			"template":     "",
+			"diameterUnit": "cm",
+		},
+	})
+
+	err := controller.CreateObject("/Physical/BASIC/A/R1/A01", models.GENERIC, map[string]any{
+		"attributes": map[string]any{
+			"rotation":     []float64{0, 0, 0},
+			"size":         []float64{1, 1, 1},
+			"posXYZ":       []float64{1, 1, 1},
+			"posXYUnit":    "m",
+			"shape":        "cube",
+			"type":         "box",
+			"diameterUnit": "cm",
+		},
+	})
+	assert.Nil(t, err)
+
+	return controller, mockAPI, mockOgree3D
+}
+
+func interactLabelTestSetup(t *testing.T) (cmd.Controller, *mocks.APIPort, *mocks.Ogree3DPort) {
+	controller, mockAPI, mockOgree3D := interactTestSetup(t)
+
+	mockGetObject(mockAPI, map[string]any{
+		"category": "rack",
+		"children": []any{chassis, rackGroup, pdu},
+		"id":       "BASIC.A.R1.A01",
+		"name":     "A01",
+		"parentId": "BASIC.A.R1",
+		"attributes": map[string]any{
+			"category": "rack",
+			"children": []any{chassis, rackGroup, pdu},
+			"id":       "BASIC.A.R1.A01",
+			"name":     "A01",
+			"parentId": "BASIC.A.R1",
+		},
+	})
+
+	return controller, mockAPI, mockOgree3D
+}
+
+func TestLabelNotStringReturnsError(t *testing.T) {
+	err := cmd.C.InteractObject("/Physical/BASIC/A/R1", "label", 1, false)
```

Comment:
Why not change the expected value of the third argument to string so the compiler checks that for us?